### DISCUSSION TITLE
Problem with table extraction

### DISCUSSION THREAD

**Discussion by IrinaOganezova** [jsvine/pdfplumber#589] [Ask for help with specific PDFs]
The lines have different length, the extraction of pdf with following parameters

{"vertical_strategy": "lines", 
                                             "horizontal_strategy": "text",
                                             "intersection_tolerance": 15,
                                             "snap_tolerance": 3,}

results is 

0  None    24  
1  None        

<a href="https://ibb.co/sb2wKR1"><img src="https://i.ibb.co/YP3kcQd/original-Table.jpg" alt="original-Table" border="0"></a>

Please help with solution

**Comment by IrinaOganezova** [comment]
https://www.dgo.gov.pt/execucaoorcamental/SintesedaExecucaoOrcamentalMensal/2006/Janeiro/0106-bol.pdf 

page 23

**Comment by jsvine** [comment]
Hi @IrinaOganezova, and thanks for providing the PDF. Parsing this in an automated way might be tricky (though possible, if you programmatically identify the location of the columns and headers), but a more manual approach like this could work:

```python
table_settings = {
    "horizontal_strategy": "text",
    "vertical_strategy": "explicit",
    "explicit_vertical_lines": [ 70, 280, 325, 370, 410, 450, 488 ]
}
``` 

That will identify a table that contains your data — but which you might also want to clean up by first using `page.crop(...)`:

![image](https://user-images.githubusercontent.com/534702/151270799-ef3742e6-022a-4ccc-8b24-5dcca99258fc.png)

**Comment by IrinaOganezova** [comment]
Thank you very much for help !
